dali/python/backend_impl.cc

```diff
@@ -1075,13 +1075,49 @@ PYBIND11_MODULE(backend_impl, m) {
         "show_tensors"_a = false,
         "show_ids"_a = false,
         "use_colors"_a = false)
-    .def("epoch_size", &Pipeline::EpochSize)
+    .def("epoch_size",  [](Pipeline* p, bool with_padding) {
+          return p->EpochSize(with_padding);
+        })
     .def("epoch_size",
-        [](Pipeline* p, const std::string& op_name) {
-          std::map<std::string, Index> sizes = p->EpochSize();
+        [](Pipeline* p, const std::string& op_name, bool with_padding) {
+          std::map<std::string, Index> sizes = p->EpochSize(with_padding);
           DALI_ENFORCE(sizes.find(op_name) != sizes.end(),
               "Operator " + op_name + " does not expose valid epoch size.");
           return sizes[op_name];
+        })
+    .def("shards_number", &Pipeline::NumberOfShards)
+    .def("shards_number",
+        [](Pipeline* p, const std::string& op_name) {
+          std::map<std::string, int> shards = p->NumberOfShards();
+          DALI_ENFORCE(shards.find(op_name) != shards.end(),
+              "Operator " + op_name + " does not expose valid number of shards.");
+          return shards[op_name];
+        })
+    .def("shard_id", &Pipeline::ShardId)
+    .def("shard_id",
+        [](Pipeline* p, const std::string& op_name) {
+          std::map<std::string, int> ids = p->ShardId();
+          DALI_ENFORCE(ids.find(op_name) != ids.end(),
+              "Operator " + op_name + " does not expose valid shard id.");
+          return ids[op_name];
+        })
+    .def("IsPadLastBatch", &Pipeline::IsPadLastBatch)
```

Comment:
```suggestion
    .def("is_pad_last_batch", &Pipeline::IsPadLastBatch)
```
I think snake case was fine, my previous comment was to keep the `is_` naming in both python and C++ API

Reply:
Fixed